cmd/groups-auto-updater/main_test.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"testing"
+
+	v1 "github.com/openshift/api/user/v1"
+	"github.com/openshift/client-go/user/clientset/versioned/fake"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+	coretesting "k8s.io/client-go/testing"
+)
+
+const (
+	configFile      = "test_files/test_config.yaml"
```

Comment:
This looks like an integration test, but written in Go. IMO we should have two types of tests:

1. unit tests, in Go, here, that verify that we call the client correctly or do filtering correctly or move data around correctly (implementation details that are important to get right but not important _how_)
2. integration tests, in shell, that use the binary as a black box and ensure the output is correct

Reply:
Yea I wanted to write a low-level unit test here. But I will remove the config, therefore the tests can be simplified as well.